### PR TITLE
FE 2 - Tried to complete according to requirements

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ const authUser = asyncHandler(async (req, res) => {
       id: user.id,
       email: user.email,
       role: user.role,
+      expiresIn: 1000 * 60 * 10,
```
Comment (vokanred) [inline backend/controllers/userController.js]: Use value from ENV

File: frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: '',
+    pathMatch: 'full',
+    redirectTo: '/todo',
+  },
+  // {
+  //   path: '**',
```
Comment (vokanred) [inline frontend/src/app/app-routing.module.ts]: Why have you commented this?

File: frontend/src/app/app.config.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { environment } from 'src/environments/environment';
+
+export const APP_CONFIG = {
```
Comment (vokanred) [inline frontend/src/app/app.config.ts:3]: As for me APP_CONFIG is an extra layer of abstraction. environment is enough

File: frontend/src/app/auth/auth-form/auth-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    email: new FormControl('', [Validators.required]),
+    password: new FormControl('', [Validators.required]),
+  });
+
+  subs: Subscription = new Subscription();
+
+  constructor(private authService: AuthService, private router: Router) {}
+
+  ngOnInit() {
+    if (this.authService.isLoggedIn()) {
```
Comment (vokanred) [inline frontend/src/app/auth/auth-form/auth-form.component.ts]: Pls create a new guard that will protect login page for authorized users (move this line to the guard)

File: frontend/src/app/auth/auth.guard.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(private authService: AuthService, private router: Router) {}
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ):
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree>
+    | boolean
+    | UrlTree {
+    if (this.authService.isLoggedIn()) {
```
Comment (vokanred) [inline frontend/src/app/auth/auth.guard.ts]: You can simplify the statements
`return this.authService.isLoggedIn() || this.router.createUrlTree(['login']);`

File: frontend/src/app/auth/auth.module.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  },
+];
+
+@NgModule({
+  declarations: [AuthFormComponent],
+  imports: [
+    CommonModule,
+    SharedModule,
+    ReactiveFormsModule,
+    RouterModule.forChild(routes),
```
Comment (vokanred) [inline frontend/src/app/auth/auth.module.ts]: Don't use forChild here.
Pls move login route to app.routing.module

File: frontend/src/app/auth/services/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  randomCheck() {
+    return this.http.post(`${APP_CONFIG.api.url}/random`, {
+      test: 'test',
+    });
+  }
+
+  isLoggedIn(): boolean {
+    return new Date().getTime() < Number(localStorage.getItem('expires_at'));
+  }
+  //I know that I should be using a backend for this, but I'm not sure how to do it yet.
```
Comment (vokanred) [inline frontend/src/app/auth/services/auth.service.ts:27]: You can handle 401 error
https://dev.to/ayeletdn/angular-http-interceptor-888

**Review by vokanred — CHANGES_REQUESTED** [review]
Pls review comments and make changes.
And you shouldn't create a new PR. Update this PR. When you create a new PR conversation is lost

**Comment by ghost** [comment]
@vokanred tried to apply suggestions by you, please check

File: frontend/src/app/auth/login.guard.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ):
+    | Observable<boolean | UrlTree>
+    | Promise<boolean | UrlTree>
+    | boolean
+    | UrlTree {
+    if (this.authService.isLoggedIn()) {
+      this._location.back();
```
Comment (vokanred) [inline frontend/src/app/auth/login.guard.ts]: I think it's OK to navigate to '/'

File: frontend/src/app/auth/auth-form/auth-form.component.ts (hunk 4 above)
Comment (vokanred) [reply frontend/src/app/auth/auth-form/auth-form.component.ts]: You already have interceptor so you can remove this logic

File: frontend/src/app/auth/auth.interceptor.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    request = request.clone({
+      setHeaders: {
+        Authorization: String(jwt),
+      },
+    });
+
+    return next.handle(request).pipe(
+      catchError((response: HttpErrorResponse) => {
+        if (response.status === 401) {
+          console.log(response);
```
Comment (vokanred) [inline frontend/src/app/auth/auth.interceptor.ts]: Clear token here (call `authService.logout()`)

**Review by vokanred — CHANGES_REQUESTED** [review]
Good job, almost done

File: frontend/src/app/auth/auth-form/auth-form.component.ts (hunk 4 above)
Comment (ghost) [reply frontend/src/app/auth/auth-form/auth-form.component.ts]: Yes, you are right, seems I forgot to remove this, thank you

**Comment by ghost** [comment]
Changed code, according to instructions, please check

**Review by vokanred — APPROVED** [review]
Done

**Pull request merged**